modules/auto_verify/plugins/auto_verify.php

```diff
@@ -73,7 +78,10 @@ function auto_verify_scheduled_task($last_run_date, $db) {
     AND delta.record_status='C' AND delta.record_substatus IS NULL
         AND delta.created_on >= TO_TIMESTAMP('$oldestRecordCreatedDateToProcess', 'DD/MM/YYYY')
         AND (($autoVerifyNullIdDiff=false AND cts.identification_difficulty IS NOT NULL AND cts.identification_difficulty<=s.auto_accept_max_difficulty)
-        OR ($autoVerifyNullIdDiff=true AND (cts.identification_difficulty IS NULL OR cts.identification_difficulty<=s.auto_accept_max_difficulty)))";
+        OR ($autoVerifyNullIdDiff=true AND (cts.identification_difficulty IS NULL OR cts.identification_difficulty<=s.auto_accept_max_difficulty)))
+    AND (s.auto_accept_taxa_filters is null
+      OR delta.taxon_meaning_id = ANY (s.auto_accept_taxa_filters)
```

Comment:
As delta.taxon_path should include delta.taxon_meaning_id in the array, I think this line of the query is unnecessary and adds complexity. However, I can imagine the scenario where it allows the query planner to do a quicker simple index check on the taxon meaning ID, before doing a more complex index check on the array contents which might be slower, only if the first check fails.
I don't think I could predict if this possible performance benefit would occur in real life without testing the query against real data.

Reply:
Thanks @johnvanbreda - it didn't occur to me that the current taxon_meaning_id would be in the taxon_path. I've modified the SQL to account for that. Best keep it as simple as possible - I imagine the array intersection is pretty efficient.